src/util/BitSet.h

```diff
@@ -0,0 +1,252 @@
+#pragma once
+
+// Copyright 2019 Stellar Development Foundation and contributors. Licensed
+// under the Apache License, Version 2.0. See the COPYING file at the root
+// of this distribution or at http://www.apache.org/licenses/LICENSE-2.0
+
+// C++ value-semantic / convenience wrapper around C bitset_t
+
+#include <memory>
+#include <ostream>
+#include <set>
+
+extern "C" {
+#include "cbitset.h"
+};
+
+class BitSet
+{
+    mutable bool mCountDirty = {true};
+    mutable size_t mCount = {0};
+    std::unique_ptr<bitset_t, decltype(&bitset_free)> mPtr;
+
+  public:
+    BitSet() : mPtr(bitset_create(), &bitset_free)
+    {
+    }
+    BitSet(size_t n) : mPtr(bitset_create_with_capacity(n), &bitset_free)
+    {
+    }
+    BitSet(std::set<size_t> const& s)
+        : mPtr(bitset_create_with_capacity(s.empty() ? 0 : *s.end()),
+               &bitset_free)
+    {
+        for (auto i : s)
+            set(i);
+    }
+    BitSet(BitSet const& other)
+        : mPtr(bitset_copy(other.mPtr.get()), &bitset_free)
+    {
+    }
+    BitSet&
+    operator=(BitSet const& other)
+    {
+        mPtr = decltype(mPtr)(bitset_copy(other.mPtr.get()), &bitset_free);
+        mCount = other.mCount;
+        mCountDirty = other.mCountDirty;
+        return *this;
+    }
+    BitSet(BitSet&& other) = default;
+    BitSet& operator=(BitSet&& other) = default;
+
+    bool
+    operator==(BitSet const& other) const
+    {
+        return bitset_equal(mPtr.get(), other.mPtr.get());
+    }
+
+    bool
+    is_subseteq(BitSet const& other) const
+    {
+        return bitset_subseteq(mPtr.get(), other.mPtr.get());
+    }
+
+    bool
+    operator<=(BitSet const& other) const
+    {
+        return is_subseteq(other);
+    }
+
+    size_t
+    size() const
+    {
+        return bitset_size_in_bits(mPtr.get());
+    }
+    void
+    set(size_t i)
+    {
+        bitset_set(mPtr.get(), i);
+        mCountDirty = true;
+    }
+    void
+    unset(size_t i)
+    {
+        bitset_unset(mPtr.get(), i);
+        mCountDirty = true;
+    }
+    bool
+    get(size_t i) const
+    {
+        return bitset_get(mPtr.get(), i);
+    }
+    void
+    clear()
+    {
+        return bitset_clear(mPtr.get());
+        mCount = 0;
```

Comment:
dead code, won't be executed

Reply:
yeah, I caught and fixed this today already, but nice catch!